conans/client/tools/system_pm.py

```diff
@@ -9,19 +9,54 @@
 class SystemPackageTool(object):
 
     def __init__(self, runner=None, os_info=None, tool=None, recommends=False):
-        env_sudo = os.environ.get("CONAN_SYSREQUIRES_SUDO", None)
-        self._sudo = (env_sudo != "False" and env_sudo != "0")
-        if env_sudo is None and os.name == 'posix' and os.geteuid() == 0:
-            self._sudo = False
-        if env_sudo is None and os.name == 'nt':
-            self._sudo = False
         os_info = os_info or OSInfo()
         self._is_up_to_date = False
         self._tool = tool or self._create_tool(os_info)
-        self._tool._sudo_str = "sudo " if self._sudo else ""
+        self._tool._sudo_str = "sudo " if self._is_sudo_enabled() else ""
         self._tool._runner = runner or ConanRunner()
         self._tool._recommends = recommends
 
+    @staticmethod
+    def _is_sudo_enabled():
+        env_sudo = os.environ.get("CONAN_SYSREQUIRES_SUDO", None)
+        if env_sudo is None and os.name == 'posix' and os.geteuid() == 0:
+            return False
+        if env_sudo is None and os.name == 'nt':
+            return False
+        return env_sudo != "False" and env_sudo != "0"
+
+    @staticmethod
+    def _check_sysrequires_mode(packages):
+        """
+        Check the working mode and return True or False if the install can progress and print in _global_output
+        a report of packages to be installed if modes are verify or disabled.
+
+        Allowed modes are (enabled, verify, disabled).
+
+        For verify mode raise a conan exception
+
+        :param packages: list of packages to be installed
+        :return: True if install can progress or False if not.
+        """
+        allowed_modes= ("enabled", "verify", "disabled")
+        mode = os.environ.get("CONAN_SYSREQUIRES_MODE", "enabled")
+        mode_to_lower = mode.lower()
+        if mode_to_lower not in allowed_modes:
+            raise ConanException("CONAN_SYSREQUIRES_MODE=%s is not allowed, allowed modes=%r" % (mode, allowed_modes))
+
+        if mode_to_lower == "enabled":
+            return True
+
+        # Report to output packages needs to be installed
+        packages_output = '\n'.join(packages)
```

Comment:
I think this function should be only `_get_sysrequire_mode()`, it could raise if invalid mode. 

Then, directly in the install() function:

```python
if mode == "disabled":  # Do not install anything
    return
elif mode == "verify" and not self._installed(packages): 
    raise "Aborted due to CONAN_SYSREQUIRES_MODE=%s and packages needs to be installed" % mode

```

Reply:
Yes it sounds ok to me. I did it in that way to avoid install method know about statuses but I can change it, no problem

Reply:
@lasote do you want to print the packages that should be installed too in case `disabled` or `verify`? it was something mentioned on the issue and addressed in PR.